### PR TITLE
exit loop when gpu job is killed

### PR DESCRIPTION
**Pull request by jgd5**
When killed is caleld, the loop will exit  on the next message or timeout (20 seconds).

### CODE REVIEW

File: training/gpu_training_job.js [hunk 1]
```diff
@@ -73,7 +73,9 @@ module.exports = class GPUTrainingJob extends BaseTrainingJob {
         try {
             await this._processResponse();
         } finally {
-            await this._scaleGPUNode(0);
+            if (this._killed !== true) {
+              await this._scaleGPUNode(0);
+            }
```
Comment (gcampax) [inline training/gpu_training_job.js]: Why this change? Even if the job was killed we still need to scale down, right?

Comment (jgd5) [reply training/gpu_training_job.js]: this._scaleGPUNode(0) iscalled when the job is killed.  This avoids calling eskctl scale twice. Although the op is idempotent, I haven't tested two concurrent eskctl scale calls.

**Comment by jgd5** [comment]
Fixes #408
